docs/DeveloperGuide.md

```diff
@@ -221,47 +216,61 @@ When the `EventCreateCommand` finishes executing, the updated `EventStorage` is
+
+<box type="info" seamless>
+
+**Note:** Any command that changes the volunteers state or events state will trigger the `saveNewState()` method for both `VersionedVolunteerStorage` and `VersionedEventStorage`. Similarly, a `undo` or `redo` command will trigger the `undo()` and `redo()` method for both `VersionedVolunteerStorage` and `VersionedEventStorage`. To avoid diagram repetition, I will mainly refer to `VersionedVolunteerStorage` in the following illustrations. However, take note that whatever happens at `VersionedVolunteerStorage` occurs for `VersionedEventStorage` as well.
+
+</box>
 
 Given below is an example usage scenario and how the undo/redo mechanism behaves at each step.
 
-Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the initial address book state, and the `currentStatePointer` pointing to that single address book state.
+Step 1. The user launches the application for the first time. The `VersionedVolunteerStorage` will be initialized with the initial volunteers state, and the `currentStatePointer` pointing to that `versionedVolunteers` state. Similarly, the `VersionedEventStorage` will be initialized with the initial events state, and the `currentStatePointer` pointing to that `versionedEvents` state.
 
 <puml src="diagrams/UndoRedoState0.puml" alt="UndoRedoState0" />
 
-Step 2. The user executes `delete 5` command to delete the 5th volunteer in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
+Step 2. The user executes `vdelete 5` command to delete the 5th volunteer in the volunteer list. The `vdelete` command calls `Model#commitToBothVersionedStorages()`, causing the modified state of the volunteer list after the `vdelete 5` command executes to be saved in the `versionedVolunteers`, and the `currentStatePointer` is shifted to the newly inserted `versionedVolunteers` state. Similarly, a new state of the event list will be saved in the `versionedEvents` (although this new state is identical to the previous events state), and the `currentStatePointer` is shifted to the newly inserted `versionedEvents` state.
 
 <puml src="diagrams/UndoRedoState1.puml" alt="UndoRedoState1" />
 
-Step 3. The user executes `add n/David …​` to add a new volunteer. The `add` command also calls `Model#commitAddressBook()`, causing another modified address book state to be saved into the `addressBookStateList`.
+Step 3. The user executes `vcreate n/David …​` to add a new volunteer. The `vcreate` command also calls `Model#commitToBothVersionedStorages()`, causing another modified volunteers state to be saved into the `versionedVolunteers`. Similarly, another events state is saved into `versionedEvents`.
```

Comment:
perhaps can improve on clarity a bit:
- `volunteers state` -> `volunteer storage state`
- `events state` -> `event storage state`

Reply:
but this just a suggestion not sure how your one completely works

Reply:
hmm thanks for suggestion, but I think mine is ok